docs/book/src/plugins/extending/extending_cli_features_and_plugins.md

```diff
@@ -0,0 +1,411 @@
+the `create api` and `create webhook` subcommands.
+
+### Plugin Subcommands
+
+Plugins are responsible for implementing the code that will be executed when the sub-commands are called.
+You can create a new plugin by implementing the [Plugin interface][plugin-interface].
+
+On top of being a `Base`, a plugin should also implement the [`SubcommandMetadata`][plugin-subc]
+interface so it can be run with a CLI. It optionally to set custom help
+text for the target  command; this method can be a no-op, which will
+preserve the default help text set by the [cobra][cobra] command
+constructors.
+
+Kubebuilder CLI plugins wrap scaffolding and CLI features in conveniently packaged Go types that are executed by the
+`kubebuilder` binary, or any binary which imports them. More specifically, a plugin configures the execution of one
+of the following CLI commands:
+
+- `init`: Initializes the project structure.
+- `create api`: Scaffolds a new API and controller.
+- `create webhook`: Scaffolds a new webhook.
+- `edit`: edit the project structure.
+
+Here’s an example of using the `init` subcommand with a custom plugin:
+
+```sh
+kubebuilder init --plugins=mylanguage.kubebuilder.io/v1
+```
+
+This would initialize a project using the `mylanguage` plugin.
+
+### Plugin Keys
+
+Plugins are identified by a key of the form `<name>/<version>`.
+There are two ways to specify a plugin to run:
+
+- Setting `kubebuilder init --plugins=<plugin key>`, which will initialize a project configured for plugin with key
+  `<plugin key>`.
+
+- A `layout: <plugin key>` in the scaffolded [PROJECT configuration file][project-file]. Commands (except for `init`, which scaffolds this file) will look at this value before running to choose which plugin to run.
+
+By default, `<plugin key>` will be `go.kubebuilder.io/vX`, where `X` is some integer.
+
+For a full implementation example, check out Kubebuilder's native [`go.kubebuilder.io`][kb-go-plugin] plugin.
+
+### Plugin naming
+
+Plugin names must be DNS1123 labels and should be fully qualified, i.e. they have a suffix like
+`.example.com`. For example, the base Go scaffold used with `kubebuilder` commands has name `go.kubebuilder.io`.
+Qualified names prevent conflicts between plugin names; both `go.kubebuilder.io` and `go.example.com` can both scaffold
+Go code and can be specified by a user.
+
+### Plugin versioning
+
+A plugin's `Version()` method returns a [`plugin.Version`][plugin-version-type] object containing an integer value
+and optionally a stage string of either "alpha" or "beta". The integer denotes the current version of a plugin.
+Two different integer values between versions of plugins indicate that the two plugins are incompatible. The stage
+string denotes plugin stability:
+
+- `alpha`: should be used for plugins that are frequently changed and may break between uses.
+- `beta`: should be used for plugins that are only changed in minor ways, ex. bug fixes.
+
+### Boilerplates
```

Comment:
Boilerplate made me think of this file https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v4/scaffolds/internal/templates/hack/boilerplate.go

Would be better to call this section templates?

Reply:
That problem with use the hack/boilerplate here is that we would need to explain what is , how is use and etc.. So, I think keep the go.mod seems good enough as a basic example

Reply:
oh no no

I am ok using `go.mod` as the example

Would change just the name of the section, because of the confusion I made

Reply:
Bollerplate is how we call any "template logic" used to do the scaffolds